src/robusta/integrations/mattermost/sender.py

```diff
@@ -0,0 +1,183 @@
+from itertools import chain
+
+import requests
+import base64
+
+from ...core.model.env_vars import ROBUSTA_LOGO_URL
+from ...core.reporting.base import *
+from ...core.reporting.blocks import *
+from ...core.reporting.utils import add_pngs_for_all_svgs
+from ...core.sinks.transformer import Transformer
+
+extension_regex = re.compile(r"\.[a-z]+$")
+MattermostBlock = Dict[str, Any]
+SEVERITY_EMOJI_MAP = {
+    FindingSeverity.HIGH: ":red_circle:",
+    FindingSeverity.MEDIUM: ":large_orange_circle:",
+    FindingSeverity.LOW: ":large_yellow_circle:",
+    FindingSeverity.INFO: ":large_green_circle:",
+}
+SEVERITY_COLOR_MAP = {
+    FindingSeverity.HIGH: "#d11818",
+    FindingSeverity.MEDIUM: "#e48301",
+    FindingSeverity.LOW: "#ffdc06",
+    FindingSeverity.INFO: "#05aa01",
+}
+SUPPORTED_IMAGES_FORMATS = ['png', 'gif', 'jpg', 'jpeg', 'bmp']
+MAX_BLOCK_CHARS = 16383  # Max allowed characters for mattermost
+
+
+class MattermostSender:
+    def __init__(
+            self, url: str, cluster_name: str, channel: Optional[str]
+    ):
+        """
+        Set the Mattermost webhook url.
+        """
+        self.url = url
+        self.cluster_name = cluster_name
+        self.channel = channel
+
+    @classmethod
+    def __add_severity_icon(cls, title: str, severity: FindingSeverity) -> str:
+        icon = SEVERITY_EMOJI_MAP.get(severity, "")
+        return f"{icon} {severity.name} - {title}"
+
+    def __format_final_message(self, mattermost_blocks: List[str],
+                               header_block: str, attachment_blocks: List[str],
+                               msg_color: str) -> Dict:
+        attachments = [{
+            "title": header_block,
+            "text": "\n".join(mattermost_blocks),
+            "color": msg_color
+        }]
+        attachments.extend([{"image_url": attachment, "color": msg_color} for attachment in attachment_blocks])
+        msg = {
+            "username": "Robusta",
+            "icon_url": ROBUSTA_LOGO_URL,
+            "attachments": attachments
+        }
+        if self.channel:
+            msg['channel'] = self.channel
+
+        return msg
+
+    def __to_mattermost(self, block: BaseBlock, sink_name: str) -> Optional[str]:
+        if isinstance(block, MarkdownBlock):
+            return Transformer.to_github_markdown(block.text)
+        elif isinstance(block, FileBlock):
+            extension = re.findall(extension_regex, block.filename)
+            if not extension:
+                return ""
+            extension = extension[0][1:]
+            if extension not in SUPPORTED_IMAGES_FORMATS:
```

Comment:
I think we ignore all text files here, like .txt, .log, .yaml
Maybe we need to check if the suffix is an image, and if so to send it as an image.
If it's not, send it as a text file attachment.

WDYT?

Reply:
The thing is, Mattermost does not support sending files attachments.
The only way we can do it at this moment - hosting a Robusta bot, similar we do with slack. Should we switch to that solution?

Reply:
The crash logs is the feature users likes the most.
So I think it's important to add it.
What do you mean by hosting a Robusta bot?